webservices/resources/legal.py

```diff
@@ -30,21 +71,32 @@ def get(self, q, from_hit=0, hits_returned=20, _type='all', **kwargs):
         else:
             types = [_type]
 
+        parsed_query = parse_query_string(q)
+        terms = parsed_query.get('terms')
+        phrases = parsed_query.get('phrases')
+
         results = {}
         total_count = 0
         for _type in types:
             query = {"query": {"bool": {
-                     "must": [{"match": {"_all": q}}, {"term": {"_type": _type}}],
-                               "should": [{"match": {"no": q}},
-                                               {"match_phrase": {"_all": {"query": q,
-                                                                          "slop": 50}
-                                                                 }
-                                                }]
+                     "must": [
+                         {"term": {"_type": _type}},
+                         ],
+                     "should": [
+                         {"match": {"no": q}},
+                         {"match_phrase": {"_all": {"query": q, "slop": 50}}},
+                         ]
                      }},
                 "highlight": {"fields": [{"text": {}},
                     {"name": {}}, {"number": {}}]},
                 "_source": {"exclude": "text"}}
 
+            if len(terms):
+                query['query']['bool']['must'].append({"match": {"_all": ' '.join(terms)}})
+
+            for phrase in phrases:
+                query['query']['bool']['must'].append({"match_phrase": {"text": phrase}})
```

Comment:
Hmm, I played around with `term`, but that behavior didn't seem right. For example, `"matters"` returned results, while `"Matters"` returned no results. `"electronic filings"` also returned no results for some reason.

As far as I can tell, `match_phrase` seems to be doing the right thing. The behavior I see matches the [elasticsearch docs](https://www.elastic.co/guide/en/elasticsearch/guide/1.x/_phrase_search.html)
> ["rock climbing"] will match only employee records that contain both “rock” and “climbing” and that display the words next to each other in the phrase “rock climbing.”

There is one quirk though, the highlights sometimes includes partial phrases which I suppose is expected given the explanation above. However, the documents returned still always match the entire phrase. In other words, searching for a phrase `"electronic filing"` returns a subset of the search `electronic filing`. Also, if you reverse the terms like `"filing electronic"`, you get no results as expected. So I'm not really sure how to proceed here. If there are configuration options for the highlights, then I feel better about the `match_phrase` approach.

Not sure if you saw my comment https://github.com/18F/fec-eregs/issues/192#issuecomment-241160861 but I wonder if the Simple Query String query will actually just do everything we want.

I pushed the current (with `match_phrase`) code up to the `feature` env, so you can play with it here https://fec-feature-proxy.18f.gov/data/legal/search/?search_type=all&search=%22electronic+filing%22